xdsl/interpreters/experimental/wgsl_printer.py

```diff
@@ -0,0 +1,248 @@
+from typing import IO, cast
+
+from xdsl.dialects import arith, builtin, gpu, memref
+from xdsl.dialects.memref import MemRefType
+from xdsl.ir import Operation, SSAValue
+from xdsl.ir.core import Attribute
+
+
+class WGSLPrinter:
+    name_dict: dict[SSAValue, str] = dict()
+    count = 0
+
+    def wgsl_name(self, v: SSAValue):
+        if v not in self.name_dict.keys():
+            if v.name_hint is not None:
+                self.name_dict[v] = f"v{v.name_hint}"
+            else:
+                self.name_dict[v] = f"v{self.count}"
+                self.count += 1
+        return self.name_dict[v]
+
+    @singledispatchmethod
+    def print(self, op: Operation, out_stream: IO[str]) -> None:
+        raise NotImplementedError(
+            f"Printing the {op.name} to WGSL is not implemented yet."
+        )
+
+    @print.register
+    def _(self, op: gpu.ModuleOp, out_stream: IO[str]):
+        for o in op.body.ops:
+            if isinstance(o, gpu.FuncOp):
+                self.print(o, out_stream)
+
+    @print.register
+    def _(self, op: gpu.FuncOp, out_stream: IO[str]):
+        print(f"Thats a gpu func : {op}")
+        for arg in op.body.block.args:
+            auth = "read"
+            arg_type = ""
+            for use in arg.uses:
+                if isinstance(use.operation, memref.Store):
+                    auth = "read_write"
+            if isinstance(arg.type, builtin.Float32Type):
+                arg_type = "f32"
+            elif isinstance(arg.type, builtin.IndexType):
+                arg_type = "u32"
+            elif isinstance(arg.type, MemRefType):
+                memref_typ = cast(MemRefType[Attribute], arg.type)
+                arg_type = f"array<{memref_typ.element_type}>"
+            arguments = f"""
+    @group(0) @binding({arg.index})
+    var<storage,{auth}> {self.wgsl_name(arg)}: {arg_type};
+"""
+            out_stream.write(arguments)
+
+        out_stream.write(
+            """
+    @compute
+    @workgroup_size(1)
+    fn main(@builtin(global_invocation_id) global_invocation_id : vec3<u32>,
+    @builtin(workgroup_id) workgroup_id : vec3<u32>,
+    @builtin(local_invocation_id) local_invocation_id : vec3<u32>,
+    @builtin(num_workgroups) num_workgroups : vec3<u32>) {
+"""
+        )
+        for operation in op.body.ops:
+            self.print(operation, out_stream)
+        out_stream.write(
+            """
+            }
+            """
+        )
+
+    @print.register
+    def _(self, op: gpu.ReturnOp, out_stream: IO[str]):
+        pass
+
+    @print.register
+    def _(self, op: gpu.BlockIdOp, out_stream: IO[str]):
+        dim = str(op.dimension.value.param).replace('"', "")
+        name_hint = self.wgsl_name(op.result)
+        out_stream.write(
+            f"""
+        let {name_hint}: u32 = workgroup_id.{dim};"""
+        )
+
+    @print.register
+    def _(self, op: gpu.ThreadIdOp, out_stream: IO[str]):
+        dim = str(op.dimension.value.param).replace('"', "")
```

Comment:
you can use `.strip('"')` instead, it may be a bit more robust? idk. Feels more "proper".